Rr/milestone 01 account setting

rupalirajput2597 opened the pull request:
## What does this PR do?
Implements the Settings section with the following options:
✅ Account Settings
✅ Notification Settings
UI for the settings page is implemented.

### Related Issues & PRs

- Create and Implement Account Settings [#28](https://github.com/baragaun/first-spark-app/issues/28)

#### Key Decisions
<!-- Explain the technical decisions and alternative approaches considered for your new implementations and refactored solutions -->
- **Implemeneted Approach**:
- Created a Settings page with two sections: Account Settings & Notification Settings.
- Keeping both settings sections in separate pages but opted for a single page with tabs for better UX.

## Breaking Changes

### Limitations & Concerns
<!-- Describe any limitations, potential edge cases, or situations where your solution might break -->
- **Known limitations**:
- Currently, only UI is implemented—functionality for saving changes is not included in this PR.
- **Potential risks**:
- Future functionality changes might require state management adjustments.

---

### Screenshots
| Description | Screenshot |

old UI:
https://github.com/user-attachments/assets/d98ad5d9-ef40-4883-b351-cf82f4eeaff0

updated UI as per provided screenshots in the issue [#28](https://github.com/baragaun/first-spark-app/issues/28):

https://github.com/user-attachments/assets/0829d1b7-af13-4a6c-a440-156e11daf2c2



---

### Reviewers
<!-- Tag at least 2 others for review -->
- [x] Reviewer 1: @jonahtwalker 
- [x] Reviewer 2: @shadowbrush 
- [x] Reviewer 3: @jahanvi-baragaun 
- [x] Reviewer 4: @RaghvindYadav
- [x] Reviewer 5: @rashmit530 


### Next Steps
- Implement the backend functionality to save and update settings.

### How to Test
2. Test steps:

- Navigate to the Settings page.
- Verify that Account Settings and Notification Settings tabs are visible.
- Check if the UI is responsive and displays correctly on different screen sizes.
3. Expected results:
- Verify that Account Settings and Notification Settings tabs are visible.
- The UI should be functional.
## Final Checklist
- [x] Linked related issues
- [x] Rebased with target branch
- [x] No lint errors
- [x] QA'd in multiple browsers
- [x] Added/updated tests (if applicable)
- [x] Mobile responsive (if applicable)
- [x] Updated documentation (if applicable)
- [x] Notified the Team of this updates readiness

---

## AI Usage
- [x] AI tools were used in development
  - **Tool used**:
  - Augment Code
  - **Prompts used**:
    - **Initial prompt**: 
    - please analyse whole project and also analyse svelte-coding-instructions.txt file carefull, we have to add more functionalities to our app using these instructions
    - I hope you have analysed svelte-coding-instructions.txt file carefully and learnt my preferences while writting svelte code
    - great, now tell me did you find "Setting option " in app's sidebar?
    - I wants to add a Setting page , which will be empty for now on  selection of setting option
    - could you please highlight the selection of sidebar options, while coding please follow svelte-coding-instruction.txt file carefully
    - great, now I want you to implement Account-Setting for me with the following features:
         * I can navigate to the notification settings
         * I can remove an email from my account
         * I can sign out of all devices with a single action
         * I can change my password
         * I can delete my account
         * I can download all of my data
remember while coding please follow svelte-coding-instruction.txt file carefully

- great, now I want to check, have we implemented basic Notification Preferences where user can select "email, push notification, none" for all events
- But I can see notification preferences already implemented
- 1.Keep the current simple implementation
- No I do not even want save functionality for now, keep it empty
- similary I want to implement basic UI for the following notification settings options with existing installed UI coponets
        * I can subscribe to the event "I received a chat message
        * I can subscribe to the event "My buddy approved my connection request"
        * I can subscribe to the event "My friend received a chat message"
        * I can subscribe to the event "Pathfinder accepted connection request"
remember while coding please follow svelte-coding-instruction.txt file carefully
- we are not using '$lib/components/ui/select' remove it
- can we keep the  Notification Preferences as a separate setting section from account setting? on Setting page
- not this way, I want clear separation, like keep taggle buttons at top as Account setting, Notification prefferences. by defailt select Account setting page. remenber while coding please follow svelte-coding-instruction.txt file carefully and  use existing installed UI components
- great, please remove notification preference option from account setting option

## References
<!-- List any helpful references used during development -->
- **Documentation**:
- **Discussions**:
- **Related PRs**:

Commit:
Feature: Account setting page

Commit:
added notifications-setting option along with account setting

shadowbrush commented:
Please remove [package-lock.json](https://github.com/baragaun/first-spark-app/pull/34/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519). We switched to `pnpm`.

shadowbrush commented:
I added the import sorting package and ran the formatter

rupalirajput2597 commented:
Hello @shadowbrush , 
- I have formatted the code 
- removed [package-lock.json](https://github.com/baragaun/first-spark-app/pull/34/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519) file 
- started using `pnpm`
- rebased the branch with `milestone 01` branch and resolved conflicts

please re-review and let me know if any changes needed. Thank you!

jonahtwalker commented:
There is an issue for this: https://github.com/baragaun/first-spark-app/issues/28

Please include these in the description instead of `N/A.` We are using Reddit as a template for our base app, so please review their UX, our user stories and the included screen shots to build this UI.

Commit:
updated the UI of account-seeting and notification setting as per th provided UI screenshots

Commit:
added dialog to edit account setting, 2.fix lint issues, 3.formatd code properly

jonahtwalker commented on src/lib/components/settings-ui/notification-settings.svelte:
![Screenshot 2025-03-11 at 12 47 31 PM](https://github.com/user-attachments/assets/1ee54b71-2e90-427f-a829-34311590822a)

This looks great! Let's reduce the options to this until we get these features in Milestone 02 & Milestone 03 👍🏼

rupalirajput2597 replied:
Sure, will comment out the rest of the code for future use

rupalirajput2597 replied:
Done!

jonahtwalker commented on src/lib/components/settings-ui/account-settings.svelte:
![Screenshot 2025-03-11 at 12 49 09 PM](https://github.com/user-attachments/assets/23e4b668-751e-48b4-9464-f76bae7acffc)

This looks great, too! Please reduce the features for now until we've got them in the next couple Milestones. They can be commented out for now. 👍🏼

rupalirajput2597 replied:
Got it, will comment out the rest of code and for future purpose. Thanks Jonah!

rupalirajput2597 replied:
Done!

jonahtwalker reviewed (changes requested):
This looks great Rupali! There's a bit of cleanup to make sure we only offer what we already support, and some further UI changes to make it more uniform. Please let me know if I can clarify anything else. 👍🏼  Excited to get this in, we're comin' close to a core app! 🚀 

Thanks again!!

### Delete Account:
- add the current email as the input placeholder
![Screenshot 2025-03-11 at 12 50 51 PM](https://github.com/user-attachments/assets/19f0731f-b769-4630-8eb4-89bfaf027c12)

### Email Addresses
- Change to `Email`
- Update this screen, as we require passwords, to look just like the Username popover
![Screenshot 2025-03-11 at 12 52 38 PM](https://github.com/user-attachments/assets/a528c51a-2a29-40b4-befe-c806582e8df1)
    - When the user submits, they then get sent the link and should see the second part
![Screenshot 2025-03-11 at 12 55 38 PM](https://github.com/user-attachments/assets/4f97bfc0-ae98-4188-a3fb-217b9aab1003)

### Password
- This should not be the reset password flow
- This should not navigate away, it should open a popover like the rest of the settings
  - The form should look just like the username and email forms
    - There should be a second input for updating the password
    - There are user stories for the Password Input and Update Password modules

Commit:
worked on review comments

rupalirajput2597 commented:
> This looks great Rupali! There's a bit of cleanup to make sure we only offer what we already support, and some further UI changes to make it more uniform. Please let me know if I can clarify anything else. 👍🏼 Excited to get this in, we're comin' close to a core app! 🚀
> 
> Thanks again!!
> 
> ### Delete Account:
> * add the current email as the input placeholder
>   ![Screenshot 2025-03-11 at 12 50 51 PM](https://private-user-images.githubusercontent.com/64275677/421549391-19f0731f-b769-4630-8eb4-89bfaf027c12.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3NTUzMzksIm5iZiI6MTc0MTc1NTAzOSwicGF0aCI6Ii82NDI3NTY3Ny80MjE1NDkzOTEtMTlmMDczMWYtYjc2OS00NjMwLThlYjQtODliZmFmMDI3YzEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEyVDA0NTAzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1ZmJhMzVkMzJlZDU2ZjA5OWNhZDFjZDM4NTQxNmVhYjIwODNiOTY4ODIyYmI1N2M4M2U2MDI5NTY5NjE3N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fRXriL7MsgR6B5R5VWs3u5dOXC8O2kSKkikYdQfGWl0)
> 
> ### Email Addresses
> * Change to `Email`
> * Update this screen, as we require passwords, to look just like the Username popover
>   ![Screenshot 2025-03-11 at 12 52 38 PM](https://private-user-images.githubusercontent.com/64275677/421550167-a528c51a-2a29-40b4-befe-c806582e8df1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3NTUzMzksIm5iZiI6MTc0MTc1NTAzOSwicGF0aCI6Ii82NDI3NTY3Ny80MjE1NTAxNjctYTUyOGM1MWEtMmEyOS00MGI0LWJlZmUtYzgwNjU4MmU4ZGYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEyVDA0NTAzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3MGU0NTA3YTViNmI2MzgxNjdjY2FmZDcwN2E4YzNkOWZlZTE3Nzk1ODRjMzUxMmJlYmExODU4ZjFiNTRjNGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qjM-_DUx-14VRDiv41rjCg0JgQy4WkCBaw41Xp6WF7A)
>   
>   * When the user submits, they then get sent the link and should see the second part
>     ![Screenshot 2025-03-11 at 12 55 38 PM](https://private-user-images.githubusercontent.com/64275677/421551562-4f97bfc0-ae98-4188-a3fb-217b9aab1003.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3NTUzMzksIm5iZiI6MTc0MTc1NTAzOSwicGF0aCI6Ii82NDI3NTY3Ny80MjE1NTE1NjItNGY5N2JmYzAtYWU5OC00MTg4LWEzZmItMjE3YjlhYWIxMDAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEyVDA0NTAzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyZTMyNWVkMDk3OWFjMTUxNzU5OGFjMGM0NmU4YTllOWE4MjA3ZDMwNjU0ZDQxY2UwY2ViMjliZDA3MTQ4Y2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.x8KrcktMll1hV9WnJXmmIeujAMSAuRHghZ0cxiScxxM)
> 
> ### Password
> * This should not be the reset password flow
> * This should not navigate away, it should open a popover like the rest of the settings
>   
>   * The form should look just like the username and email forms
>     
>     * There should be a second input for updating the password
>     * There are user stories for the Password Input and Update Password modules

Hello @jonahtwalker , I have worked on review comments and pushed the changes, here is the screencast for the improved version of setting feature.

https://github.com/user-attachments/assets/9c9c78b8-3cff-4b5c-95fc-115c300cbe64


And here is the screen recording for password validation same as signup password validation:

https://github.com/user-attachments/assets/3d8b0936-6250-49bb-b8d1-8a30c1c327fb

Commit:
matched password validation with signup password validation

jonahtwalker reviewed (approved):
Thank you Rupali, let's get this in! I'm going to create some follow up issues for specific Modules that we can improve as we get them integrated with the node-client. 👍🏼
